backends/p4tools/modules/testgen/targets/bmv2/test/small-step/p4_asserts_parser_test.cpp

```diff
@@ -56,12 +57,11 @@ ConstraintsVector loadExample(const char *curFile, bool flag) {
     auto *originalEnv = getenv("P4C_16_INCLUDE_PATH");
     setenv("P4C_16_INCLUDE_PATH", includeDir.c_str(), 1);
     const IR::P4Program *program = nullptr;
-    options.file = cstring(sourcePath);
-    options.file += curFile;
-    if (access(options.file, 0) != 0) {
+    options.file = std::filesystem::path(sourcePath) / curFile;
+    if (access(options.file.c_str(), 0) != 0) {
         // Subpath for bf-p4c-compilers.
-        options.file = cstring(sourcePath);
-        options.file += curFile;
+        // FIXME: what is the logic behind here?
```

Comment:
I assume this code was written to also work with compilers with a slightly different structure, e.g. the bf-p4c. But not sure what the approach here was there is likely an simpler solution.

Reply:
Well, the thing is: the code inside the `access` condition is equivalent to the code above: we just emitting `sourcePath / curFile`. It looks like some leftover from previous changes / refactoring.

Reply:
Hopefully yes, but I just do not know the story behind this code, so maybe something was lost at some point of time.

Reply:
The oldest commit I can find for this file is #3568. It was at a different point in the tree then and was later moved in a cleanup.

The duplicated code in the `if (access(...))` block was present then, so it's nothing that we've accidentally corrupted after it was added to the p4c codebase.

Reply:
Yeah feel free to be aggressive with refactorings here. This code is eventually meant to go away anyway since the parser is replaced with the actual [p4-constaints](https://github.com/p4lang/p4-constraints) parser.